src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs

```diff
@@ -154,6 +154,57 @@ private MethodIL TryGetPerInstantiationIntrinsicMethodIL(MethodDesc method)
                             return ComparerIntrinsics.EmitEqualityComparerCreate(method);
                     }
                     break;
+                case "ComparerHelpers":
+                    {
+                        if (owningType.Namespace != "Internal.IntrinsicSupport")
+                            return null;
+
+                        if (methodName == "EnumOnlyCompare")
+                        {
+                            //calls CompareTo for underlyingType to avoid boxing
+
+                            TypeDesc elementType = method.Instantiation[0];
+                            if (!elementType.IsEnum)
+                                return null;
+
+                            ILOpcode convInstruction;
+                            int typeSize = ((DefType)elementType).InstanceFieldSize.AsInt;
+                            if (typeSize <= 4)
+                            {
+                                convInstruction = ILOpcode.conv_i4;
```

Comment:
This should not need any conversions. It should just load the two arguments and call the underlying type CompareTo like: `ldarga 0, ldarg 1, call ..., ret`.